# Description: Add a pid_file option so it can be specified via commandline
# (for init scripts) rather than splitting config files for this one option.
# The daemonizing method doesn't properly create PIDFILE specified in the 
# init scripts and without this creates pid files in /tmp. This is fixed
# upstream in ticket CHEF-1022.
Index: chef-0.8.6/chef-solr/lib/chef/solr/application/indexer.rb
===================================================================
--- chef-0.8.6.orig/chef-solr/lib/chef/solr/application/indexer.rb	2010-03-07 00:30:07.000000000 -0700
+++ chef-0.8.6/chef-solr/lib/chef/solr/application/indexer.rb	2010-03-07 00:30:17.000000000 -0700
@@ -75,6 +75,12 @@
           :description => "Daemonize the process",
           :proc => lambda { |p| true }
 
+        option :pid_file,
+          :short => "-P PIDFILE",
+          :long => "--pidfile PIDFILE",
+          :description => "PID file to use.",
+          :proc => nil
+
         option :amqp_host,
           :long => "--amqp-host HOST",
           :description => "The amqp host"
Index: chef-0.8.6/chef-solr/lib/chef/solr/application/solr.rb
===================================================================
--- chef-0.8.6.orig/chef-solr/lib/chef/solr/application/solr.rb	2010-03-07 00:29:26.000000000 -0700
+++ chef-0.8.6/chef-solr/lib/chef/solr/application/solr.rb	2010-03-07 00:30:02.000000000 -0700
@@ -72,6 +72,12 @@
           :description => "Daemonize the process",
           :proc => lambda { |p| true }
 
+        option :pid_file,
+          :short => "-P PIDFILE",
+          :long => "--pidfile PIDFILE",
+          :description => "PID file to use.",
+          :proc => nil
+
         option :solr_jetty_path,
           :short => "-W PATH",
           :long => "--solr-jetty-dir PATH",
